### DISCUSSION TITLE
Simple C++ pub/sub app in tutorial not working

### DISCUSSION THREAD

**Discussion by flyotlin** [eProsima/Fast-DDS#2743] [Q&A] [answered]
Hi guys, I ran into some problems and hope if it's a right place to ask question here.

I followed the [tutorial](https://fast-dds.docs.eprosima.com/en/latest/fastdds/getting_started/simple_app/simple_app.html) to write a simple pub/sub app in C++, but as I opened up both pub and sub from different terminals, none of which starts doing what they're supposed to do.

I ran my app in a docker container on Macbook Air M1 (macOS BigSur v11.6.2).

The docker container base image is [Docker Fast DDS 2.6.0](https://www.eprosima.com/index.php/component/ars/repository/eprosima-fast-dds/eprosima-fast-dds-2-6-0/ubuntu-fastdds-v2-6-0-tar?format=raw).

Is there any hints or thoughts on this problem? Anything would be appreciated!

**Comment by flyotlin** [comment]
**Some Updates:**
I tried a clean installation from [binaries](https://eprosima.com/index.php/component/ars/repository/eprosima-fast-dds/eprosima-fast-dds-2-6-0) in another Ubuntu 21.04 computer, and followed the tutorial exactly the same as the above.

It turned out working well, maybe the problem's related to Docker Image or M1 Macbook Air.

**Comment by MiguelCompany** [comment, marked as answer]
As stated in the [Docker for Apple documentation](https://docs.docker.com/desktop/mac/apple-silicon/#known-issues):

```
In summary, running Intel-based containers on Arm-based machines should be regarded as “best effort” only.
We recommend running arm64 containers on Apple silicon machines whenever possible
```

The docker image for Fast DDS is Intel-based, so there may be issues when running it on Apple silicon.
